<launch>
    <arg name="rviz_mode" default="true"/>
    
    <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find dorna_description)/urdf/dorna.urdf.xacro' stand_alone:=true" />

    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="false" output="screen">
        <remap from="/joint_states" to="/dorna/robot_info/joint_states" />
    </node>

    <node name="dorna" pkg="dorna_ros" type="dorna_ros_node.py" output="screen"/>

    <node if="$(arg rviz_mode)" name="rviz" pkg="rviz" type="rviz" args="-d $(find dorna_ros)/config/rviz/dorna_rviz_config.rviz"/>
</launch>
